x-pack/plugin/rollup/qa/rest/src/javaRestTest/java/org/elasticsearch/rollup/RollupSearchIT.java

```diff
@@ -0,0 +1,172 @@
+        String firstGenerationRollupIndex = ".rollup-" + firstGenerationIndex;
+        assertBusy(() -> assertThat(indexExists(DataStream.getDefaultBackingIndexName(dataStream, 2)), is(true)), 30, TimeUnit.SECONDS);
+        rollupIndex(client(), firstGenerationIndex, firstGenerationRollupIndex, "{\n" +
+            "  \"groups\" : {\n" +
+            "    \"date_histogram\": {\n" +
+            "      \"field\": \"@timestamp\",\n" +
+            "      \"calendar_interval\": \"1M\"\n" +
+            "    },\n" +
+            "    \"terms\": {\n" +
+            "      \"fields\": [\"units\"]\n" +
+            "    }\n" +
+            "  },\n" +
+            "  \"metrics\": [\n" +
+            "    {\n" +
+            "      \"field\": \"temperature\",\n" +
+            "      \"metrics\": [\"max\", \"sum\", \"avg\"]\n" +
+            "    }\n" +
+            "  ]\n" +
+            "}");
+        String query = "{\n" +
+            "  \"size\": 0,\n" +
+            "  \"aggs\": {\n" +
+            "      \"monthly_temperatures\": {\n" +
+            "          \"date_histogram\": {\n" +
+            "              \"field\": \"@timestamp\",\n" +
+            "              \"calendar_interval\": \"1M\"\n" +
+            "          },\n" +
+            "          \"aggs\": {\n" +
+            "              \"avg_temperature\": {\n" +
+            "                  \"avg\": {\n" +
+            "                      \"field\": \"temperature\"\n" +
+            "                  }\n" +
+            "              },\n" +
+            "              \"max_temperature\": {\n" +
+            "                  \"max\": {\n" +
+            "                      \"field\": \"temperature\"\n" +
+            "                  }\n" +
+            "              }\n" +
+            "          }\n" +
+            "      }\n" +
+            "  }\n" +
+            "}";
+        Map<String, Object> response = search(dataStream, query);
+        Map<String, Object> shards = (Map<String, Object>) response.get("_shards");
+        assertThat(shards.get("skipped"), equalTo(1));
+        Map<String, Object> aggs = (Map<String, Object>) response.get("aggregations");
+        Map<String, Object> monthlyTemps = (Map<String, Object>) aggs.get("monthly_temperatures");
+        List<Map<String, Object>> buckets = (List<Map<String, Object>>) monthlyTemps.get("buckets");
+        assertThat(buckets.size(), equalTo(2));
+        assertThat(buckets.get(0).get("doc_count"), equalTo(2));
+        assertEquals(28.1, (Double) ((Map<String, Object>) buckets.get(0).get("max_temperature")).get("value"), 0.0001);
+        assertEquals(27.8, (Double) ((Map<String, Object>) buckets.get(0).get("avg_temperature")).get("value"), 0.0001);
+        assertThat(buckets.get(1).get("doc_count"), equalTo(2));
+
+        // Index new doc so that we confirm merging live and rollup indices
+        indexDocument(client(), dataStream,
```

Comment:
Hey @talevy there seems to be some kind of bug discovered here. 
After running rollover and rollup at the datastream, it seems that the rollup index becomes the write index for the datastream. 

So when indexing another document, test fails with the following exception:

```
org.elasticsearch.client.ResponseException: method [POST], host [http://127.0.0.1:57070], URI [logs-foo/_doc?refresh], status line [HTTP/1.1 403 Forbidden]
    {"error":{"root_cause":[{"type":"cluster_block_exception","reason":"index [.rollup-.ds-logs-foo-2021.01.27-000001] blocked by: [FORBIDDEN/8/index write (api)];"}],"type":"cluster_block_exception","reason":"index [.rollup-.ds-logs-foo-2021.01.27-000001] blocked by: [FORBIDDEN/8/index write (api)];"},"status":403}
```

Reproduce with: 

```
 ./gradlew ':x-pack:plugin:rollup:qa:rest:javaRestTest' --tests "org.elasticsearch.rollup.RollupSearchIT.testDatastreamRollupSearch" -Dtests.seed=C88D0F54446CAD0D -Dtests.security.manager=true -Dtests.locale=id -Dtests.timezone=Africa/Banjul -Druntime.java=15

Reply:
fixed in https://github.com/elastic/elasticsearch/pull/67783/commits/6f563f1f4c16ca00f8dfe3be60023c1ba0e3b836!